jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/caws/DevEnvStatusWatcher.kt

```
@@ -0,0 +1,112 @@
+        val projectName = System.getenv(CawsConstants.CAWS_ENV_PROJECT_NAME_VAR) ?: error("project env var null")
+        val client = connection.awsClient<CodeCatalystClient>()
+        val coroutineScope = projectCoroutineScope(project)
+        coroutineScope.launch(getCoroutineBgContext()) {
+            val initialEnv = client.getDevEnvironment {
+                it.id(envId)
+                it.spaceName(org)
+                it.projectName(projectName)
+            }
+            val inactivityTimeout = initialEnv.inactivityTimeoutMinutes()
+            if (inactivityTimeout == 0) {
+                LOG.info("Dev environment inactivity timeout is 0, not monitoring")
+                return@launch
+            }
+            val inactivityTimeoutInSeconds = inactivityTimeout * 60
+
+            // ensure the JetBrains inactivity tracker and the activity api are in sync
+            val jbActivityStatusJson = UnattendedStatusUtil.getStatus()
+            val jbActivityStatus = jbActivityStatusJson.projects?.first()?.secondsSinceLastControllerActivity ?: 0
+            notifyBackendOfActivity((System.currentTimeMillis() - (jbActivityStatus * 1000)).toString())
+            var secondsSinceLastControllerActivity = jbActivityStatus
+
+            while (true) {
+                val statusJson = UnattendedStatusUtil.getStatus()
+                val lastActivityTime = statusJson.projects?.first()?.secondsSinceLastControllerActivity ?: 0
+                if (lastActivityTime < secondsSinceLastControllerActivity) {
+                    // update the API in case of any activity
+                    notifyBackendOfActivity((System.currentTimeMillis() - (lastActivityTime * 1000)).toString())
+                }
+                secondsSinceLastControllerActivity = lastActivityTime
+
+                val lastRecordedActivityTime = CawsEnvironmentClient.getInstance().getActivity().timestamp ?: System.currentTimeMillis().toString()
```

might be a bad idea to silently ignore here

---

Added a log statement and `return` if we don't have a response from the api